libbeat/reader/multiline/multiline.go

```diff
@@ -78,6 +79,20 @@ func New(
 	separator string,
 	maxBytes int,
 	config *Config,
+) (*Reader, error) {
+	if config.Type == patternMode || config.Type == nil {
+		return newMultilinePatternReader(r, separator, maxBytes, config)
+	} else if config.Type == countMode {
+		return newMultilineCountReader(r, separator, maxBytes, config)
```

Comment:
I do not think it is needed to introduce a complete new reader. Some small modifications to the existing multline-reader will do exactly what is requested.

Reply:
It is not introducing a new type. Only the validation of the config options is separated.